linera-core/src/updater.rs

```diff
@@ -210,17 +209,64 @@ where
     A: ValidatorNode + Clone + 'static,
     S: Storage + Clone + Send + Sync + 'static,
 {
-    async fn send_certificate<T: 'static + ProcessableCertificate>(
```

Comment:
Keeping this code factorized on the client side seems preferrable. In principle, the way we can do it is by including a generic `dispatch` method in `ProcessableCertificate`: `cert.dispatch(f_confirmed, f_validated, f_timeout)` calls `f_confirmed(cert)` if `cert` is a confirmed certificate, etc.

Reply:
@ma2bd  is [Reintroduce generic send_certificate method](https://github.com/linera-io/linera-protocol/pull/2996/commits/2645b494cf16cece3692650aaaad07e3da11e236) what you meant?

Reply:
It's also causing the `web` [CI job to fail](https://github.com/linera-io/linera-protocol/actions/runs/12146314774/job/33869940822) due to new bounds required by the trait to be executed in async context in Wasm.

If you're OK with that, I'd prefer to revert this commit entirely.

Reply:
On the other hand, I just split this into two methods on main anyway. :innocent: